xtable-core/src/main/java/org/apache/xtable/delta/DeltaSchemaExtractor.java

```diff
@@ -172,7 +186,11 @@ private InternalSchema toInternalSchema(
         type = InternalType.DOUBLE;
         break;
       case "binary":
-        type = InternalType.BYTES;
+        if ("uuid".equals(originalMetadata.getString("xtableLogicalType"))) {
```

Comment:
the `getString` is failing when I run locally. I think there needs to be a `contains` check first or else you get this `java.util.NoSuchElementException: key not found: xtableLogicalType`

Reply:
> the `getString` is failing when I run locally. I think there needs to be a `contains` check first or else you get this `java.util.NoSuchElementException: key not found: xtableLogicalType`

Sorry, I have committed this changes locally before but forget to push, already fixed in latest commit